### DISCUSSION TITLE
2nd cy.wait fails when there are two cy.intercept with same alias

### DISCUSSION THREAD

**Discussion by varunpradhan** [cypress-io/cypress#16052] [Questions and Help]
### Current behavior
There is a cy.intercept which has a dynamic URL with an alias. When we call cy.wait for this alias for the second time, then it fails to look for the new URL/alias.

As cy.intercept URL is dynamic, 2 different routes are generated with same alias:

![image](https://user-images.githubusercontent.com/26091432/115145420-0a53a500-a06f-11eb-96d2-e935df98b863.png)

Now the 1st cy.wait is passed:

![image](https://user-images.githubusercontent.com/26091432/115145445-340ccc00-a06f-11eb-8b40-b20f9cc9e76b.png)

When system tries to wait on 2nd cy.wait it fails:

![image](https://user-images.githubusercontent.com/26091432/115145472-5bfc2f80-a06f-11eb-9351-67b6b93e335d.png)

Below is stack printed on Cypress Test Runner:

![image](https://user-images.githubusercontent.com/26091432/115145503-6e766900-a06f-11eb-92e7-621fd1d9be56.png)

### Desired behavior
System should catch the latest URL/alias and wait for it.

### Test code to reproduce
I have the git repo which we can clone - **https://github.com/varunpradhan/bddeapp**

Run the test file - **cypress\integration\defaultCy\multipleCalls.js**

Below is the cy.intercept code which has dynamic URL, which is also found under **cypress\step_definitions\common\editEmployee.js** in the git repo

```
class editEmployee {
    benefitEmp(index) {
        cy.intercept({
            method: "GET",
            url: `**/Benefit/Details/${index}`,
        }).as('empBenefit');

        cy.get(`[href="/Benefit/Details/${index}"]`).click()

        cy.wait('@empBenefit').then(({ response }) => {
            expect(response.statusCode).to.equal(200);
        })
    }

}
```

### Versions

* **Cypress version**: 6.8.0
* **Node version**: 12.18.4

**Comment by varunpradhan** [comment]
@bahmutov Can you please help me with this issue?

**Reply by CinkadeusBG** [reply]
I experienced the same issue.  My solution was to use unique names for my intercepts.
